src/core/Table/Table.tsx

```diff
@@ -267,11 +274,29 @@ export const Table = <
         maxWidth: 48,
         columnClassName: 'iui-slot',
         cellClassName: 'iui-slot',
-        Header: ({ getToggleAllRowsSelectedProps }: HeaderProps<T>) => (
-          <Checkbox {...getToggleAllRowsSelectedProps()} />
-        ),
+        Header: ({ getToggleAllRowsSelectedProps }: HeaderProps<T>) => {
+          const onChange = () => {
+            const isAllEnabledRowsSelected =
+              instance.isAllRowsSelected ||
+              instance.rows.every(
+                (row) =>
+                  instance.state.selectedRowIds[row.id] ||
+                  isRowDisabled(row.original),
+              );
+            instance.toggleAllRowsSelected(!isAllEnabledRowsSelected);
+          };
+          return (
+            <Checkbox
+              {...getToggleAllRowsSelectedProps()}
```

Comment:
Wouldnt from here also come onChange callback, which you need to merge?

Reply:
I want to override it and did it here.

Reply:
What if user has something added for his custom stuff? Or its not possible..

Reply:
What you mean by saying custom? It is controlled by us and comes from react-table.

Reply:
I meant, if user cannot pass same prop on row himself?

Reply:
It is controlled completely by us, so user can't do anything here.